index.js

```diff
@@ -69,6 +69,9 @@ function promisedHandlebars (Handlebars, options) {
   // Re-register all built-in-helpers to ensure that their methods are wrapped
   engine.registerHelper(engine.helpers)
 
+  // Preserve synchronous 'compile' function
+  engine.compileSync = engine.compile.bind({})
```

Comment:
The empty object is the `thisArg`. I guess it doesn't really matter what you pass in this case since `this` is not being accessed by the [compile function](https://github.com/wycats/handlebars.js/blob/master/lib/handlebars/compiler/compiler.js)